cloud/aws/actuators/cluster/actuator.go

```diff
@@ -26,7 +26,7 @@ import (
 )
 
 type ec2Svc interface {
-	ReconcileNetwork(*providerconfigv1.Network) error
+	ReconcileNetwork(*clusterv1.Cluster, *providerconfigv1.Network) error
```

Comment:
Should we pass the cluster here? We could leverage `NewService(...)` to pass in more information and store the cluster name in the Service struct, it seems the information is being used as a context.

Reply:
Yes, but this requires a larger refactoring, which I would prefer to keep out of this PR. WDYT?

Reply:
I'm fine with deferring a larger refactoring. For a stopgap, I would definitely prefer that we limit it to just the attributes of the cluster that we need rather than the entire cluster object.